Xamarin.Forms.Build.Tasks/BuildException.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Xml;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Forms.Xaml;
+
+namespace Xamarin.Forms.Build.Tasks
+{
+	[Serializable]
+	class BuildException : Exception
+	{
+		public BuildExceptionCode Code { get; private set; }
+
+		public IXmlLineInfo XmlInfo { get; private set; }
+
+		public string[] MessageArgs { get; private set; }
+
+		public override string HelpLink { get => Code.HelpLink; set => base.HelpLink = value; }
+
+		public BuildException(BuildExceptionCode code, IXmlLineInfo xmlInfo, Exception innerException, params object[] args)
+			: base(FormatMessage(code, xmlInfo, args), innerException)
+		{
+			Code = code;
+			XmlInfo = xmlInfo;
+			MessageArgs = args?.Select(a=>a.ToString()).ToArray();
```

Comment:
this is necessary to the exception can be serialized

Reply:
And then I assume the args array itself will never have a null object in it?

Reply:
good catch. fixed